script.js

```diff
@@ -1,4 +1,251 @@
+    currentPlayerNum += 1
+    firstDiceNum = generateDiceNumber(6).toString();
+    secondDiceNum = generateDiceNumber(6).toString();
+
+    output = generateNormalPickDiceOrderMessage();
+    CURRENT_GAME_STATE = userPickNumberOrder
+  } else if (CURRENT_GAME_STATE === userPickNumberOrder) {
+    if (!['1', '2'].includes(userInput)) return "Please submit only 1 or 2.";
+
+    let numberPicked = firstDiceNum + secondDiceNum;
+    if (userInput === '2') {
+      numberPicked = secondDiceNum + firstDiceNum;
+    }
+    playerChoiceNumbers.push(numberPicked);
+
+    output = generateNumberPickedMessage();
+    CURRENT_GAME_STATE = computerDiceNumberGenerated;
+
+    if (currentPlayerNum === playerAmount) {
+      CURRENT_GAME_STATE = calculateResult;
+    }
+  }
+  if (CURRENT_GAME_STATE === calculateResult) {
+    output = generateLeaderboard();
+    resetGame();
+  }
+  return output;
+}
+
+const resetGame = function () {
+  CURRENT_GAME_STATE = rollDice;
+  currentPlayerNum = 0;
+  playerChoiceNumbers = [];
+};
+
+// DICE GENERATOR
+const generateDiceNumber = (diceNum) => {
+  return ~~(Math.random() * diceNum) + 1;
+};
+
+// MESSAGE GENERATORS BELOW
+const generateLeaderboard = function () {
+  let output = '';
+  let tieArray = [];
+  // Borrowed from StackOverflow. (I was googling how to sort and getting the result by index then 3 mins afterwards Stackoverflow went down :joy:)
+  const orderedIndex = Array.from(Array(playerChoiceNumbers.length).keys())
+    .sort((a, b) => playerChoiceNumbers[a] > playerChoiceNumbers[b] ? -1 : (playerChoiceNumbers[b] > playerChoiceNumbers[a]) | 0);
+  if (CALCULATE_BY === LOWEST_MODE) {
+    orderedIndex.reverse();
+  }
+  const maxNum = playerChoiceNumbers[orderedIndex[0]];
+  // forEach to generate leaderboard output. I wrote a function if tie.
+  orderedIndex.forEach((index, val) => {
+    output += `Player #${val + 1} had a score of ${playerChoiceNumbers[val]}<br />`
+    if (playerChoiceNumbers[val] === maxNum && val !== orderedIndex[0]) {
+      tieArray.push(val + 1);
+    }
+  });
+  if (tieArray.length > 0) {
+    // Sort here is just for nicer input.
+    output += `<br />Players ${[...tieArray, orderedIndex[0] + 1].sort()} have won with a score of ${maxNum}! Congrats!`
+  } else {
+    output += `<br />Player #${orderedIndex[0] + 1} won with a score of ${maxNum}! Congrats!`;
+  }
+  output += "<br />Play again by pressing submit.";
+  return output;
+};
+
+const generateNormalPickDiceOrderMessage = function () {
+  return `Hello, Player #${currentPlayerNum}.<br />Your first number is: ${firstDiceNum}. Your second number is: ${secondDiceNum}.<br />Submit '1' to make your number starts with first number, or '2' to make your number starts with second number.`;
+};
+
+const generateNumberPickedMessage = function () {
+  return `You picked ${numberPicked}. Now it's Player #${currentPlayerNum + 1}'s turn. Press submit to generate your dice.`;
+};
```

Comment:
Since you are only using these outputs once, they don't have to be placed in functions are can be written directly at lines 171 and 182